### PR TITLE
Make volume ls output order

### PR DESCRIPTION
**Pull request by HackToday**
Fixes: #20384
Add order support for volume ls to make it easy
to external users to consume.

Signed-off-by: Kai Qiang Wu(Kennan) wkqwu@cn.ibm.com

### CODE REVIEW

File: integration-cli/docker_cli_volume_test.go [hunk 1]
```diff
@@ -65,19 +65,34 @@ func (s *DockerSuite) TestVolumeCliInspectMulti(c *check.C) {
...
+	dockerCmd(c, "run", "-v", "soo:"+prefix+"/foo", "busybox", "ls", "/")
 
 	out, _ = dockerCmd(c, "volume", "ls")
 	outArr := strings.Split(strings.TrimSpace(out), "\n")
 	c.Assert(len(outArr), check.Equals, 4, check.Commentf("\n%s", out))
 
-	// Since there is no guarantee of ordering of volumes, we just make sure the names are in the output
-	c.Assert(strings.Contains(out, id+"\n"), check.Equals, true)
-	c.Assert(strings.Contains(out, "test\n"), check.Equals, true)
+	assertVolList(c, out, []string{"aaa", "soo", "test"})
```
Comment (thaJeztah) [inline integration-cli/docker_cli_volume_test.go]: Can we be 100% sure that no other volumes exist from other tests? Just wondering if there's a chance another volume gets into the list, making the test become flaky

Comment (HackToday) [inline integration-cli/docker_cli_volume_test.go]: hi @thaJeztah  right now, I did not find the good reason the volumes left there, it means other test not good deal with clean up. So it seems if it happen, it means other tests not good written.

Wdyt?  If we can not assume  others test good written. what do you think the best way to test the sort behavior ?

Comment (thaJeztah) [inline integration-cli/docker_cli_volume_test.go]: I think that;

1. In this case perhaps `assertVolList` can skip volumes that are not in the list of `expectedVols`? Alternatively, only check if they are all present, and appear the expected order
2. We should make sure that (just as with containers) all volumes are removed after a test completes (but that can be a separate PR)

Comment (cpuguy83) [inline integration-cli/docker_cli_volume_test.go]: If a volume is left in the list, then there is a bug somewhere since we clean up all volumes on every test.

**Comment by cpuguy83** [comment]
Just to have the discussion... should this also be grouped by driver?

**Comment by thaJeztah** [comment]
> Just to have the discussion... should this also be grouped by driver?

Not for now, I think; we don't currently group, e.g., containers per "image" used? Being able to `--filter driver=foo` would be nice though.

**Comment by HackToday** [comment]
Yes, I agree with @thaJeztah  for related group behavior, we need use filter ways, instead of depend the ls output behavior

File: integration-cli/docker_cli_volume_test.go (hunk 1 above)
Comment (HackToday) [inline integration-cli/docker_cli_volume_test.go]: So @thaJeztah @cpuguy83  seems the above "left volume" issue is not related with this PR. as all other tests follow same logic it depends on

func (s *DockerSuite) TearDownTest(c *check.C) {
        unpauseAllContainers()
        deleteAllContainers()
        deleteAllImages()
        deleteAllVolumes()
        deleteAllNetworks()
}


That `DockerSuite` would did that, I think, we not need to call delete volume in single test

**Comment by HackToday** [comment]
ping @thaJeztah and @cpuguy83 

Could you check above comments for the test ?  Thanks

File: integration-cli/docker_cli_volume_test.go (hunk 1 above)
Comment (thaJeztah) [inline integration-cli/docker_cli_volume_test.go]: Agreed; if the TearDown removes all volumes, individual tests should not call delete volume

**Comment by vdemeester** [comment]
@cpuguy83 agreed with @thaJeztah. Let's keep it simple for now and not group by driver. Adding a `driver` filter and/or sorting options (for volume but also for other docker citizens — images, containers, …) could be done after :angel:

**Comment by cpuguy83** [comment]
We should keep this consistent with the `network ls` sorting.

**Comment by HackToday** [comment]
@cpuguy83 network in under review, it use same logic sort with name

**Comment by HackToday** [comment]
network is here https://github.com/docker/docker/pull/20383

**Comment by calavera** [comment]
LGTM

**Comment by cpuguy83** [comment]
LGTM
